management/vmutils/rolestate.go

```diff
@@ -0,0 +1,31 @@
+package vmutils
+
+import (
+	"time"
+
+	"github.com/MSOpenTech/azure-sdk-for-go/management"
+	vm "github.com/MSOpenTech/azure-sdk-for-go/management/virtualmachine"
+)
+
+// Blocks until all role instances in deployment to reach desired power state
+func WaitForDeploymentPowerState(client management.Client, cloudServiceName, deploymentName string, desiredPowerstate vm.PowerState) error {

```

Comment:
@paulmey this does not timeout or break at all, it's too dangerous for a user to call it. can you add a `timeout time.Duration`

also can you make the polling interval 30s?